src/stores/constraints.ts

```diff
@@ -52,25 +52,16 @@ export const sanitizeConstraintName = (constraintName: string) => {
 };
 
 export const constraintViolationsMap: Readable<ConstraintViolationsMap> = derived(
-  [constraintViolationsResponseMap, planStartTimeMs, constraints],
-  ([$constraintViolationsResponseMap, $planStartTimeMs, $constraints]) =>
-    Object.entries($constraintViolationsResponseMap).reduce((map, [constraintName, violations]) => {
-      // TODO remove this name translation once the constraint violations are properly stored in the DB
-      const translatedConstraintName = sanitizeConstraintName(constraintName);
-      const matchingConstraint = $constraints.find(constraint => constraint.name === translatedConstraintName);
-
-      if (matchingConstraint) {
-        map[matchingConstraint.id] = violations.map(({ associations, gaps, windows }) => ({
-          associations,
-          constraintId: matchingConstraint.id,
-          constraintName: translatedConstraintName,
-          gaps,
-          windows: windows.map(({ end, start }) => ({
-            end: $planStartTimeMs + end / 1000,
-            start: $planStartTimeMs + start / 1000,
-          })),
-        }));
-      }
+  [constraintViolationsResponse, planStartTimeMs],
```

Comment:
Like transform the windows wherever needed outside of the store? Only rub is that the transformation is needed in multiple places. Or are you suggesting that we merge `constraintViolationsResponse` and `constraintViolations` into just `constraintViolations` and transform the windows after we fetch the data?

Reply:
I modified this part of the code in my latest commit Aaron, basically now `constraintViolations` is derived and does the window transformation rather than using that intermediate map that we had before.

Reply:
Ah ok got it, looks good to me.